crates/pera-core/src/dwallet_mpc/network_dkg.rs

```diff
@@ -36,23 +48,51 @@ pub struct DwalletMPCNetworkKeyVersions {
 /// Encapsulates all the fields in a single structure for atomic access.
 pub struct DwalletMPCNetworkKeyVersionsInner {
     /// The validators' decryption key shares.
-    pub validator_decryption_key_share: HashMap<DWalletMPCNetworkKey, Vec<Vec<u8>>>,
+    pub validator_decryption_key_share: HashMap<
+        DWalletMPCNetworkKeyScheme,
+        Vec<HashMap<PartyID, <AsyncProtocol as Protocol>::DecryptionKeyShare>>,
+    >,
     /// The dWallet MPC network decryption key shares (encrypted).
     /// Map from key type to the encryption of the key version.
-    pub key_shares_versions: HashMap<DWalletMPCNetworkKey, Vec<NetworkDecryptionKeyShares>>,
+    pub key_shares_versions: HashMap<DWalletMPCNetworkKeyScheme, Vec<NetworkDecryptionKeyShares>>,
     /// The status of the network supported key types for the dWallet MPC sessions.
     pub status: DwalletMPCNetworkKeysStatus,
 }
 
 impl DwalletMPCNetworkKeyVersions {
     /// Creates a new instance of the network encryption key shares.
-    pub fn new(epoch_store: &AuthorityPerEpochStore) -> Self {
-        let decryption_key_share = epoch_store
-            .load_validator_decryption_key_shares_from_system_state()
-            .unwrap_or(HashMap::new());
+    pub fn new(
+        epoch_store: &AuthorityPerEpochStore,
+        weighted_threshold_access_structure: &WeightedThresholdAccessStructure,
+        class_groups_decryption_key: ClassGroupsDecryptionKey,
+    ) -> Self {
+        // Safe to unwrap because the authority name is always present in the epoch store.
```

Comment:
When you this kind of thing, it entails us needing to verify such complex assumptions, and I don't see the reason why not to return an error instead.

Think: an `unwrap` that fails throws an exception that halts the entire blockchain. This is a potential critical vulnerability. Auditors cannot ignore this. So we would have to go into Sui, and try to make sure under no condition this is possible - why waste our time on such useless endeavor, when you could have simply returned `Result`?